src/interface.jl

```diff
@@ -178,7 +196,15 @@ julia> collect(vertices(SimpleGraph(4)))
  4
 ```
 """
-vertices(g::AbstractGraph) = _NI("vertices")
+function vertices end
+
+@traitfn function vertices(g::AG::(!HasContiguousVertices)) where {AG <: AbstractGraph}
+    return _NI("vertices")
+end
+
+@traitfn function vertices(g::AG::(HasContiguousVertices)) where {AG <: AbstractGraph}
+    return Base.OneTo(nv(g))
+end
```

Comment:
This is beautiful.

Reply:
right? Smaller interface means easier to build new graph types for people